File: yunva17/프로그래머스/Lv1/체육복.js
const solution = (n, lost, reserve) => {
  let answer = n;

  // lost, reserve 중복 제거
  reserve.map((item) => {
    if (lost.includes(item)) {
      reserve = reserve.filter((x) => x !== item);
      lost = lost.filter((x) => x !== item);
    }
  });

  for (let i = 1; i <= n; i++) {
    if (lost.includes(i)) {
      if (reserve.includes(i - 1)) {
        reserve = reserve.filter((item) => item !== i - 1);
      } else if (reserve.includes(i + 1)) {
        reserve = reserve.filter((item) => item !== i + 1);
      } else {
        answer -= 1;
      }
    }
  }

  return answer;
};

File: pbkdpwls/구현/체육복.py
def solution(n, lost, reserve):
    answer = 0

    lost_len = len(lost)

    for i in lost:
        if i in reserve:
            reserve.remove(i)
            lost.remove(i)
            lost_len -= 1

    for i in lost:
        if (i - 1) in reserve:
            reserve.remove(i - 1)
            lost_len -= 1

        elif (i + 1) in reserve:
            reserve.remove(i + 1)
            lost_len -= 1

    answer = n - lost_len

    return answer